compiler/rustc_codegen_ssa/src/errors.rs

```diff
@@ -384,7 +387,7 @@ impl<G: EmissionGuarantee> Diagnostic<'_, G> for LinkingFailed<'_> {
                 } else if arg.as_encoded_bytes().ends_with(b".rlib") {
                     let rlib_path = Path::new(&arg);
                     let dir = rlib_path.parent().unwrap();
-                    let filename = rlib_path.file_name().unwrap().to_owned();
+                    let filename = rlib_path.file_stem().unwrap().to_owned();
```

Comment:
This now removes the `.rlib` suffix even when not replacing a crate hash. Maybe change the regex to `-[0-9a-f]+\.rlib$` instead?

Reply:
oh, excellent catch. fixed.

Reply:
I created #143482 because that's actually not the only problem (brace expansion doesn't trigger with a single element).